backend/app/readers/reader/wikipedia_reader.py

```diff
@@ -0,0 +1,37 @@
+import json
+from collections.abc import Generator
+from pathlib import Path
+
+from app.models.wikipedia_article.wikipedia_article import WikipediaArticle
+from app.readers.reader.reader import Reader
+from app.utils.encodings.encodings_list import encodings
+
+
+class WikipediaReader(Reader):
+    """A concrete implementation of Reader.
+
+    Read in Wikipedia output data and yield them as WikipediaArticles.
+    """
+
+    def __init__(self, file_path: Path) -> None:
+        self.__file_path = file_path
+
+    def read(self) -> Generator[WikipediaArticle, None, None]:
+        """Read in Wikipedia output data and yield Records."""
+
+        json_list = (
+            self.__file_path.open(mode="r", encoding="utf-8").read().strip().split("\n")
```

Comment:
Need to `with` this open.

Reply:
Parse each line and yield rather than buffering the entire file in memory.

Reply:
```
with self.__file_path.open() as file_:
   for line in file_:
       record_json = json.loads(line)
       if record_json["type"] != "RECORD":
          continue
       yield WikipediaArticle ...
```

Reply:
Yes, I am using a context manager to parse and yield Records now.